src/components/Menu/index.tsx

```diff
@@ -1,114 +1,141 @@
-      size: 16,
-      ...labelProps,
+    const handleOpen = () => {
+      // Some triggers, like dropbox, might change themselves in an open state
+      // so let's measure it after the trigger is rendered
+      setTimeout(measureTrigger);
+      onVisibleChangeWrapper(true);
     };
-    return (
-      <Pressable
-        key={label}
-        style={({ pressed }) => [styles.item, pressed && styles.itemPressed]}
-        disabled={disabled}
-        onPress={() => {
-          onSelect(label);
-          handleClose();
-        }}
-      >
-        {iconProps && (
-          <Icon
-            key={label}
-            width={20}
-            style={styles.icon}
-            color='text-primary'
-            {...iconProps}
-            disabled={disabled}
-          />
-        )}
-        <Text {...textProps} disabled={disabled}>
-          {label}
-        </Text>
-      </Pressable>
-    );
-  };
 
-  const renderItems = () => {
-    const render = customRenderItem ? customRenderItem : renderItem;
-    const items = data.map((item) => render(typeof item === 'string' ? { label: item } : item));
-    const useScroll = menuHeight > windowHeight * MENU_SCREEN_FACTOR;
-    if (useScroll) {
+    const handleClose = () => onVisibleChangeWrapper(false);
+
+    useImperativeHandle(ref, () => ({
+      open: handleOpen,
+      close: handleClose,
+    }));
+
+    const renderItem = ({ label, iconProps, labelProps, disabled }: IItemData) => {
+      const textProps: IConditionalTextProps | { size: TextStyle['fontSize'] } = {
+        size: 16,
+        ...labelProps,
+      };
       return (
-        <ScrollView style={{ maxHeight: windowHeight * MENU_SCREEN_FACTOR }}>{items}</ScrollView>
+        <Pressable
+          key={label}
+          style={({ pressed }) => [styles.item, pressed && styles.itemPressed]}
+          disabled={disabled}
+          onPress={() => {
+            onSelect(label);
+            handleClose();
+          }}
+        >
+          {iconProps && (
+            <Icon
+              key={label}
+              width={20}
+              style={styles.icon}
+              color='text-primary'
+              {...iconProps}
+              disabled={disabled}
+            />
+          )}
+          <Text {...textProps} disabled={disabled}>
+            {label}
+          </Text>
+        </Pressable>
       );
-    }
-    return items;
-  };
+    };
 
-  const renderContext = () => (
-    <Modal visible={visible} transparent animationType='none'>
-      <Pressable style={styles.overlay} onPress={handleClose} />
-      <Elevation variant='extraLight' style={styles.dropdown}>
-        {renderItems()}
-      </Elevation>
-    </Modal>
-  );
+    const renderItems = () => {
+      const render = customRenderItem ? customRenderItem : renderItem;
+      const items = data.map((item) => render(typeof item === 'string' ? { label: item } : item));
+      const useScroll = menuHeight > windowHeight * MENU_SCREEN_FACTOR;
```

Comment:
sounds like a hook)

Reply:
Yeah :) I didn't find better naming. `isScroll` or `hasScroll` does not match well.